service-base/src/main/java/org/eclipse/hono/service/http/HttpServiceBase.java

```diff
@@ -0,0 +1,461 @@
+                                             final String detail, final String contentType) {
+
+        LOG.debug("Service unavailable: {}", detail);
+        Map<CharSequence, CharSequence> headers = new HashMap<>(2);
+        headers.put(HttpHeaders.CONTENT_TYPE, contentType != null ? contentType : "text/plain");
+        headers.put(HttpHeaders.RETRY_AFTER, String.valueOf(retryAfterSeconds));
+        endWithStatus(response, HTTP_UNAVAILABLE, headers, detail, contentType);
+    }
+
+    /**
+     * Ends a response with a given HTTP status code and detail message.
+     *
+     * @param response The HTTP response to write to.
+     * @param status The status code to write to the response.
+     * @param headers HTTP headers to set on the response (may be {@code null}).
+     * @param detail The message to write to the response's body (may be {@code null}).
+     * @param contentType The content type of the message (if {@code null}, then <em>text/plain</em> is used}.
+     * @throws NullPointerException if response is {@code null}.
+     */
+    protected static void endWithStatus(final HttpServerResponse response, final int status,
+                                        Map<CharSequence, CharSequence> headers, final String detail, final String contentType) {
+
+        Objects.requireNonNull(response);
+        response.setStatusCode(status);
+        if (headers != null) {
+            for (Map.Entry<CharSequence, CharSequence> header : headers.entrySet()) {
+                response.putHeader(header.getKey(), header.getValue());
+            }
+        }
+        if (detail != null) {
+            if (contentType != null) {
+                response.putHeader(HttpHeaders.CONTENT_TYPE, contentType);
+            } else {
+                response.putHeader(HttpHeaders.CONTENT_TYPE, "text/plain");
+            }
+            response.end(detail);
+        } else {
+            response.end();
+        }
+    }
+
+    /**
+     * Gets the value of the <em>Content-Type</em> HTTP header for a request.
+     *
+     * @param ctx The routing context containing the HTTP request.
+     * @return The content type or {@code null} if the request doesn't contain a <em>Content-Type</em> header.
+     * @throws NullPointerException if context is {@code null}.
+     */
+    protected static String getContentType(final RoutingContext ctx) {
+
+        return Objects.requireNonNull(ctx).request().getHeader(HttpHeaders.CONTENT_TYPE);
+    }
+
+    /**
+     * The default port.
+     *
+     * Subclasses should override this method to provide a service specific default value.
+     * <p>
+     *
+     * @return 8443
+     */
+    @Override
+    public int getPortDefaultValue() {
+        return 8443;
+    }
+
+    /**
+     * The default insecure port.
+     *
+     * Subclasses should override this method to provide a service specific default value.
+     * <p>
+     *
+     * @return 8443
```

Comment:
Uh. My bad. That is even more embarrassing :blush: I updated it, with the same ports and fixed JavaDoc.